text/0000-better-env-handling.md

```diff
@@ -0,0 +1,89 @@
+- Start Date: 2020-04-23
+- Relevant Team(s): Ember.js, Learning
+- RFC PR: [#619](https://github.com/emberjs/rfcs/pull/619)
+- Tracking: (leave this empty)
+
+# Better environment handling
+
+## Summary
+
+This RFC proposes to add a tiny module similar to `@glimmer/env`, to allow to more easily work with common environment questions.
+
+## Motivation
+
+This RFC tries to solve two common problems:
+
+1 - **Handling test-specific code**.
+
+The old way of doing this, `Ember.testing`, is not really ideal and recommended anymore.
+
+Example use cases where you might want to use this:
+
+- Disabling polling code in tests
+- Preventing test-breaking code like `window.reload()`
+- Preventing log output in tests
+
+2 - **Handling debug-specific (=non-production) code**.
+
+You can use `import { DEBUG } from "@glimmer/env"` for this, or `import { runInDebug } from "@ember/debug"`, although the later [seems to be buggy as of now](https://github.com/emberjs/ember.js/issues/18912).
+
+An exmaple use case where you might want to use this:
+
+- Auto-signing in a mocked user in development environment
+
+This RFC proposes to provide a more more streamlined, approved and documented way to handle these cases, by introducing an `@ember/env` namespace which can be used like this:
+
+```js
+import { DEBUG, TESTING } from "@ember/env";
+```
+
+## Detailed design
+
+```js
+import { DEBUG } from "@ember/env";
+```
```

Comment:
I'd much prefer to keep with `@glimmer/env` and teach it better, than introduce another new thing. It is my opinion (personal, not an official core team position just yet) that we _really do_ want to move forward with the mental model that Ember itself is layered on top of `@glimmer/*` (very much like the "npm install your way to Ember" terminology from waaaaaay back when Glimmer.js was originally introduced), and we are actually kinda close to that these days.

If we do that, this change becomes quite a bit smaller as well (adding a `TESTING` named export to `@glimmer/env`).

Reply:
:+1: will update this accordingly!